completed screen

sergeewdw opened the pull request.

sergeewdw committed:
completed screen

sergeewdw commented:
Попробовал что такое кодекс и забыл отключить от гитхаба

spviz commented on Controlls/CollectionView + Extension.swift:
```diff
@@ -0,0 +1,71 @@
+extension UICollectionView {
+    func sectionLayoutMetric() -> NSCollectionLayoutSection {
+        let spacing: CGFloat = 10
+        let itemSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1),
```
Строки надо переносить так
```
let itemSize = NSCollectionLayoutSize(
    widthDimension: .fractionalWidth(1),
    heightDimension: .fractionalHeight(1)
)
```

spviz commented on Controlls/CollectionView + Extension.swift:
```diff
@@ -0,0 +1,71 @@
+import UIKit
+
+extension UICollectionView {
```
в целом эти лэйауты используются только в RocketViewController, лучше там сделать private extension

spviz commented on Controlls/Header.swift:
```diff
@@ -0,0 +1,52 @@
+        return label
+    }()
+
+    private let mainView: UIView = {
```
лишняя вьюха, можно добавлять сразу на contentView

spviz commented on Controlls/Header.swift:
```diff
@@ -0,0 +1,52 @@
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func configure(_ header: String) {
```
typealias HeaderViewModel = String

func configure(_ header: HeaderViewModel)

spviz commented on Controlls/LaunchesModule/View/LaunchesViewController.swift:
```diff
@@ -3,6 +3,8 @@ import UIKit
 final class LaunchesViewController: UIViewController {
     private var viewModels: [LaunchCellVIewModels] = []
     private let networkService = NetworkService()
+    private let rocketName: String
```
```
struct LaunchesViewModel {
    let launches: [LaunchCellVIewModel]
    let rocketName: String
    let rocketId: String
}
```

spviz commented on Controlls/LaunchesModule/View/LaunchesViewController.swift:
```diff
@@ -30,11 +32,20 @@ final class LaunchesViewController: UIViewController {
         return label
     }()
 
+    init(rocket: RocketInfo) {
```
init(launches: LaunchesViewModel)

spviz commented on Controlls/Model/ItemType.swift:
```diff
@@ -0,0 +1,18 @@
+    case image
+    case characteristic(SettingsType)
+    case info(CellName)
+    case stageInfo(isFirst: Bool, CellStageName)
```
а если третья ступень добавится? isFirst не сработает

spviz commented on Controlls/SettingsModule/View/SettingsViewController.swift:
```diff
@@ -101,19 +93,32 @@ private extension SettingsViewController {
     @objc
     func closeButtonTapped() {
-        dismiss(animated: true, completion: nil)
+        navigationController?.popViewController(animated: true)
```
экран модально должен появляться, у тебя сейчас он пушится. соответсвенно если модальноН, то тут должен быть dismiss

spviz commented on Controlls/View/ButtonCell.swift:
```diff
@@ -0,0 +1,53 @@
+
+final class ButtonCell: UICollectionViewCell {
+    static var identifier = "ButtonCellKey"
+    var onStartingsTap: (() -> Void)?
```
а давай тут делегат сделаем

spviz commented on Controlls/View/ButtonCell.swift:
```diff
@@ -0,0 +1,53 @@
+        return button
+    }()
+
+    private let mainView: UIView = {
```
лишняя вью

spviz replied:
И везде дальше у тебя есть эти лишние вьюхи, поправь везде, дополнительно писать не буду

spviz commented on Controlls/View/ImageCell.swift:
```diff
@@ -0,0 +1,95 @@
+
+final class ImageCell: UICollectionViewCell {
+    static var identifier = "ImageCellKey"
+    var onSettingsTap: (() -> Void)?
```
переведи на делегат

spviz commented on Controlls/View/ImageCell.swift:
```diff
@@ -0,0 +1,95 @@
+    func configure(_ rocket: RocketInfo) {
+        nameRocketLabel.text = rocket.name
+        if let url = URL(string: rocket.flickrImages?.randomElement() ?? "") {
+            self.rocketImageView.kf.setImage(with: url)
```
самое время оверраднуть prepareForReuse и сеттить self.rocketImageView.image в nil

spviz commented on Controlls/View/RocketViewController.swift:
```diff
@@ -0,0 +1,274 @@
+    var rocketInfo: RocketInfo
+    private let storage: StorageProvider
+    private lazy var collectionView: UICollectionView = makeCollectionView()
+    private var dataSource: UICollectionViewDiffableDataSource<Section, ItemType>?
```
сделать lazy и создавать прям тут, а не в ините

spviz commented on Controlls/View/RocketViewController.swift:
```diff
@@ -0,0 +1,274 @@
+import Kingfisher
+
+final class RocketViewController: UIViewController {
+    var rocketInfo: RocketInfo
```
private, через инит же присваиваем

spviz commented on Controlls/View/RocketViewController.swift:
```diff
@@ -0,0 +1,274 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        dataSource = makeDataSource()
+        applySnapshot()
```
как будто бы лишний вызов

spviz commented on Controlls/View/RocketViewController.swift:
```diff
@@ -0,0 +1,274 @@
+                    ofKind: kind, withReuseIdentifier: Header.identifier, for: indexPath
+                  ) as? Header else {
+                return UICollectionReusableView()
+            }
```
лишняя проверка и странное форматирование. 

```
guard let header = collectionView.dequeueReusableSupplementaryView(
    ofKind: kind,
    withReuseIdentifier: Header.identifier,
    for: indexPath
) as? Header else {
    return UICollectionReusableView()
}
```

spviz commented on Controlls/View/RocketViewController.swift:
```diff
@@ -0,0 +1,274 @@
+    func createLayout() -> UICollectionViewCompositionalLayout {
+        let layout = UICollectionViewCompositionalLayout { [weak self] section, _ -> NSCollectionLayoutSection? in
+            switch section {
+            case 0:
```
switch Sections.allCases[section]

sergeewdw committed:
Внес правки

Поправил что сказал из голосового и Гита, проверил темную тему и светлую, вроде нормально
Подправил еще экран настроек, цвета сигментов сделал как в ТЗ и маленькие правки внес
